### PR TITLE
Update to S3Boto3Storage

### PR DESCRIPTION
**Pull request by foadlind**
Update the storage backend to **S3Boto3Storage**.

The changes to the configuration parameters:
* `S3_ADDRESSING_STYLE` is added;
* `S3_SIGNATURE_VERSION` is added;
* `S3_REGION` is added;
* `S3_AUTO_CREATE_BUCKET` is removed;
 
Additionally, the requirement to have the bucket ACL set to public is removed.

### CODE REVIEW

**Comment by fghaas** [comment]
Looks good, but can you please squash these all into one commit?

Also, please update the commit message along the lines of <https://cbea.ms/git-commit/< and <http://www.gazlene.net/getting-work-done-in-open-source.html>. Thank you!

**Comment by fghaas** [comment]
Also, please add `Fixes: #8` to the bottom of the updated commit message. (See #8.)

File: tutors3/patches/openedx-cms-common-settings [hunk 1]
```diff
@@ -0,0 +1,2 @@
+COURSE_IMPORT_EXPORT_STORAGE = DEFAULT_FILE_STORAGE
+USER_TASKS_ARTIFACT_STORAGE = DEFAULT_FILE_STORAGE
```
Comment (fghaas) [inline tutors3/patches/openedx-cms-common-settings:2]: Missing newline at end of file.

File: setup.py [hunk 2]
```diff
@@ -8,12 +8,12 @@
     readme = f.read()
 
 setup(
-    name="tutor-s3",
+    name="tutor-contrib-s3",
```
Comment (fghaas) [inline setup.py:11]: This rename still needs a mention in the commit message.

File: tutors3/patches/openedx-common-settings [hunk 3]
```diff
@@ -1,18 +1,17 @@
...
+
+ORA2_FILEUPLOAD_BACKEND = "s3"
+FILE_UPLOAD_STORAGE_BUCKET_NAME = "{{ S3_STORAGE_BUCKET }}"
+
+AWS_S3_SIGNATURE_VERSION = "{{ S3_SIGNATURE_VERSION }}"
+AWS_S3_ENDPOINT_URL = "{{ "https" if ENABLE_HTTPS else "http" }}://{{ S3_HOST }}:{{ S3_PORT }}"
 AWS_S3_USE_SSL = {{ "True" if S3_USE_SSL else "False" }}
 AWS_S3_SECURE_URLS = {{ "True" if S3_USE_SSL else "False" }}
-AWS_S3_CALLING_FORMAT = "boto.s3.connection.OrdinaryCallingFormat"
-AWS_AUTO_CREATE_BUCKET = {{ "True" if S3_AUTO_CREATE_BUCKET else "False" }}
```
Comment (fghaas) [inline tutors3/patches/openedx-common-settings:6]: I understand that `AWS_AUTO_CREATE_BUCKET` is slated for removal from django-storages, and it defaults to `False`. What I'm not sure about is whether this is true for django-storages **1.8** (which Open edX still depends on). Even if we drop the `S3_AUTO_CREATE_BUCKET` config option, does it hurt if we explicitly set `AWS_AUTO_CREATE_BUCKET = False` here?

File: tutors3/plugin.py [hunk 4]
```diff
@@ -11,14 +11,16 @@
         "VERSION": __version__,
         "HOST": "",
```
[the remark this answers is not in the document]
Comment (foadlind) [reply tutors3/plugin.py:12]: You are right. It is not good as is. I am in favor of providing a default (`s3.amazonaws.com`).

File: tutors3/patches/openedx-common-settings (hunk 3 above)
Comment (foadlind) [reply tutors3/patches/openedx-common-settings:6]: In django-storages 1.8, `AWS_AUTO_CREATE_BUCKET` is [set to `False` by default](https://github.com/jschneier/django-storages/blob/1.8/storages/backends/s3boto.py#L210). But we can be explicit here as the [documentation for that version](https://django-storages.readthedocs.io/en/1.8/backends/amazon-S3.html) just says that is it optional and fails to mention what the default value is.

File: setup.py (hunk 2 above)
Comment (foadlind) [reply setup.py:11]: I will add it

File: tutors3/patches/openedx-cms-common-settings (hunk 1 above)
Comment (foadlind) [reply tutors3/patches/openedx-cms-common-settings:2]: Good catch :)

File: tutors3/patches/openedx-common-settings (hunk 3 above)
Comment (fghaas) [reply tutors3/patches/openedx-common-settings:6]: Yes, let's be explicit and set it to `False`. That's also [the precedent set by `tutor-minio`](https://github.com/overhangio/tutor-minio/blob/master/tutorminio/patches/openedx-common-settings#L12).

File: tutors3/plugin.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (fghaas) [reply tutors3/plugin.py:12]: Sounds good to me. :)

**Review by fghaas — APPROVED** [review]
Looks good to me now, thanks!

**Pull request merged**